src/nlmsg.rs

```diff
@@ -148,24 +148,24 @@ impl NlmsgMut {
         Self::new(buf)
     }
 
-    pub fn nested(ty: u16) -> Self {
-        // 12 => sizeof(ty) + sizeof(nested_attr_ty) + sizeof(nested_attr_data_u32)
-        let mut buffer = Self(BytesMut::with_capacity(12));
+    pub fn nested(&mut self, ty: u16) -> Self {
+        let mut nested_buffer = self.0.split_off(self.0.len());
 
         // len of NlAttr will be updated by finish_nested
-        buffer.0.put_slice(bytemuck::bytes_of(&NlAttr {
+        nested_buffer.put_slice(bytemuck::bytes_of(&NlAttr {
             len: 0,
             ty: ty | libc::NLA_F_NESTED as u16,
         }));
-        buffer
+
+        Self(nested_buffer)
     }
 
-    pub fn finish_nested(&mut self, parent: &mut NlmsgMut) {
-        let len = self.0.len();
+    pub fn finish_nested(&mut self, nested: &mut NlmsgMut) {
```

Comment:
Shouldn't this move instead of use `&mut`?
```suggestion
    pub fn finish_nested(&mut self, mut nested: NlmsgMut) {
```

Reply:
I didn't know you could this, I'm pretty new to Rust, that seems to be the right thing to do indeed. Thanks for the tip ! I've pushed a new commit for it